src/codeu/chat/client/commandline/Chat.java

```diff
@@ -361,12 +403,16 @@ public void invoke(Scanner args) {
     //
     panel.register("m-add", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String message = args.hasNext() ? args.nextLine().trim() : "";
-        if (message.length() > 0) {
-          conversation.add(message);
-        } else {
+      public void invoke(List<String> args) {
+        if(args.size() != 1){
+          System.out.println("ERROR: Failed to add new user");
+        }
+        else{
+          String message = args.get(0);
+          if (message.length() < 0) {
```

Comment:
@mbassiouny Just to clarify, we don't need the if statement within this else statement, correct? so having it as
`else{
 String message = args.get(0);
  conversation.add(message);`
will work?